### PR TITLE
Add files via upload

### LINKED ISSUE #12: Be able to receive emails.
Try to set up a crude system to receive emails.

### PR DESCRIPTION
**Pull request by wpb103**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                        filepath = os.path.join(subject, filename)
+                        # write the file
+                        open(filepath, "w").write(body)
+                        # open in the default browser
+                        webbrowser.open(filepath)
+                    print("=" * 100)
+
+                emails.append({
+                    "sender": username,
+                    "receiver": from_,
```
Comment (haleyhagler227) [inline main.py]: "sender" : from_,
"receiver" : username,

They got switched

File: main.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import imaplib
+import email
+from email.header import decode_header
+import webbrowser
+import os
+
+def receive_emails(server, username, password):
+    # account credentials
+    username = "potatomail.test@gmail.com"
+    password = "potato55"
```
Comment (haleyhagler227) [inline main.py]: These are defined in the function definition, These line should probably be removed.

File: main.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import webbrowser
+import os
+
+def receive_emails(server, username, password):
+    # account credentials
+    username = "potatomail.test@gmail.com"
+    password = "potato55"
+
+    # create an IMAP4 class with SSL
+    imap = imaplib.IMAP4_SSL("imap.gmail.com")
```
Comment (haleyhagler227) [inline main.py]: Replace "IMAP.gmial.com" with server, so this can be passed in

File: main.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    # authenticate
+    imap.login(username, password)
+
+    status, messages = imap.select("INBOX")
+    # number of top emails to fetch
+    N = 3
```
Comment (haleyhagler227) [inline main.py]: Most likely will want N passed to the function

File: main.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    # account credentials
+    username = "potatomail.test@gmail.com"
+    password = "potato55"
+
+    # create an IMAP4 class with SSL
+    imap = imaplib.IMAP4_SSL("imap.gmail.com")
+    # authenticate
+    imap.login(username, password)
+
+    status, messages = imap.select("INBOX")
```
Comment (haleyhagler227) [inline main.py]: Will want to be able to choose the inbox. should probably also be passed in

File: main.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+                emails.append({
+                    "sender": username,
+                    "receiver": from_,
+                    "subject": subject,
+                    "body": body,
+                    "id": i
+                })
+    imap.close()
+    imap.logout()
```
Comment (haleyhagler227) [inline main.py]: need to return emails at the end of the function

**Review by haleyhagler227 — CHANGES_REQUESTED** [review]
The file needs a few tweaks before this is a function that can work with the rest of the app

**Comment by wpb103** [comment]
i resolved all those issues in the function, but it won't let me reupload the file. but i have all requested changes in it

**Review by haleyhagler227 — APPROVED** [review]
These look good!

**Comment by jlc1541** [comment]
The file looks splendid, and good use of loops.

**Pull request merged**